internal/set/set_test.go

```diff
@@ -30,6 +30,17 @@ func TestAdd(t *testing.T) {
 	assert.False(t, s.Contains(3))
 }
 
+func TestAddMustNotContainDuplicates(t *testing.T) {
+	s := NewWithValues("hello")
+	assert.False(t, s.Contains("world"))
+	s.Add("world")
+	assert.True(t, s.Contains("world"))
+	s.Add("world") // Adding an already-present element
+	assert.True(t, s.Contains("world"))
+	// Must not containe `world` twice.
+	assert.Equal(t, []string{"hello", "world"}, s.Values())
```

Comment:
Non-blocking: I think this can be tested in `TestAdd` or `TestValues` (or both). OTOH this works fine, and the test file is small enough.

Reply:
done.